### ISSUE TITLE
Bug in generate_invoice_line()

### ISSUE THREAD

**Issue by garnierclement** [blawesom/osc-draft-invoicing#3]
Hi blawesom,

I think there is a bug in the function generate_invoice_line(), see traceback below.

```
Traceback (most recent call last):
  File "main.py", line 135, in <module>
    main(check_arg_dates(sys.argv[1:]))
  File "main.py", line 106, in main
    invoice_draft = create_draft_bill(icu_conn, region, invoice_draft, date_range)
  File "main.py", line 42, in create_draft_bill
    invoice_draft.append(generate_invoice_line(account_email, region, line, catalog))
  File "main.py", line 70, in generate_invoice_line
    return {'Account': account_email, 'Region': region,'Entry': key_name[7:], 'Quantity': line['Value'],'Cost': line['Value'] * generate_tinatype_price(line, region, catalog)/1000}
UnboundLocalError: local variable 'key_name' referenced before assignment
```

In this function, `key_name` is not defined in the elif clause

```
def generate_invoice_line(account_email, region, line, catalog):
    # Create line reference for search in catalog
    key = '.'.join(['unitPrice', line['Service'], line['Operation'], line['Type'], line['Zone']])
    for entry in catalog['Entries']:
        if entry['Key']==key:
            key_name = '.'.join([line['Service'], line['Operation'], line['Type']])
            return {'Account': account_email, 'Region': region,'Entry': key_name[7:], 'Quantity': line['Value'],'Cost': line['Value'] * entry['Value']/1000}
        elif line.get('Type', '').startswith('BoxUsage:tina'):
            return {'Account': account_email, 'Region': region,'Entry': key_name[7:], 'Quantity': line['Value'],'Cost': line['Value'] * generate_tinatype_price(line, region, catalog)/1000}
```

**Comment by blawesom** [comment]
Can you try out with the last commit : 33378a6944a7ba97f09c6e6e1c83171338bc567b

I should have tina entries on my own account by tomorrow to ease the debug :)

**Comment by garnierclement** [comment]
Still not working.
New issues in the function `generate_tinatype_price(line, region, catalog)` 

```
os_type = re.search('RunInstances-(\d+)-OD', line['Type']).group(1)
````

Here `Type` must be changed by `Operation`

Also the variables `core_price` and `ram_price` are not defined ...

**Comment by ihaz** [comment]
Hi blawesom,
I did a little test to check the return of the regex

```
import re
var = "FCU.RunInstances-0001-OD.BoxUsage:t2.small"
os_type = re.search('RunInstances-(\d+)-OD', var).group(1)
print('TYPE OS : {}'.format(os_type))
$ python3 testRex.py
TYPE OS : 0001
```
looks good ! 
Next step for checking the content of line['Type']

**Comment by garnierclement** [comment]
@ihaz actually the content of `line['Operation']` must be matched against the regex (and not `line['Type']`)

**Comment by blawesom** [comment]
Fixed with commit d8647b18c1fb98bb2a8bad30efd8153409efd5fe

**Comment by garnierclement** [comment]
It's working :p
